mdapi/database/main.py

```diff
@@ -43,30 +43,30 @@
 
 
 def list_branches(status="Active"):
-    urlx = "%s/api/collections?clt_status=%s" % (standard.PKGDB2_URL, status)
-    resp = requests.get(urlx, verify=standard.PKGDB2_VERIFY)
+    urlx = f"{standard.PKGDB2_URL}/api/collections?clt_status={status}"
+    resp = requests.get(urlx, verify=standard.PKGDB2_VERIFY)  # noqa : S113
     resp.raise_for_status()
     data = resp.json()
     servlogr.logrobjc.info("Branches metadata acquired.")
     return data["collections"]
 
 
 def fetch_database(name, repmdurl, location):
-    servlogr.logrobjc.info("[%s] Downloading file %s to %s" % (name, repmdurl, location))
-    resp = requests.get(repmdurl, verify=standard.DL_VERIFY)
+    servlogr.logrobjc.info(f"[{name}] Downloading file {repmdurl} to {location}")
+    resp = requests.get(repmdurl, verify=standard.DL_VERIFY)  # noqa : S113
     resp.raise_for_status()
     with open(location, "wb") as filestrm:
         filestrm.write(resp.content)
 
 
 def extract_database(name, arcvname, location):
-    servlogr.logrobjc.info("[%s] Extracting %s to %s" % (name, arcvname, location))
+    servlogr.logrobjc.info(f"[{name}] Extracting {arcvname} to {location}")
     if arcvname.endswith(".xz"):
         with lzma.open(arcvname) as inp, open(location, "wb") as otptfile:
             otptfile.write(inp.read())
     elif arcvname.endswith(".tar.gz"):
         with tarfile.open(arcvname) as tararchv:
-            tararchv.extractall(path=location, members=check_archive_validity())
+            tararchv.extractall(path=location, members=check_archive_validity())  # noqa : S202
```

Comment:
Is requiring Python >= 3.12 an option? In this case, rather use this:

```suggestion
            tararchv.extractall(path=location, filter="data")
```

Your “workaround” won’t work BTW. 🤷

Reply:
I am reminded of @dralley's [comment](https://github.com/fedora-infra/mdapi/issues/161#issuecomment-1522445080) from some eons ago and am tempted to purge those lines to see if that forsakes the codebase. If it does not, I will make the changes that you mentioned - barring to see if the `filter` option is available in Python versions lower than 3.12.

Reply:
> The workaround does work though.

Does it? The `check_archive_validity()` function returns `True` (a boolean), `.extractall()` expects a list of archive members (i.e. files) to be extracted from the tarball.

Reply:
@nphilipp the last part was a (poorly executed) sarcasm. Check the image of the image of the Ostrich for reference.

Reply:
Ah okay 😂 – sometimes I’m slow on the uptake 😉.